diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b6ac26b..77ba468 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -108,7 +108,7 @@ if  (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         LINK_FLAGS      "-nodefaultlibs ${GRAPHITE_LINK_FLAGS}"
         LINKER_LANGUAGE C)
     if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86|i.86")
-        add_definitions(-mfpmath=sse -msse2)
+        add_definitions(-mfpmath=sse)
     endif()
     if (CMAKE_COMPILER_IS_GNUCXX)
         add_definitions(-Wno-class-memaccess -Wdouble-promotion)
@@ -140,7 +140,7 @@ if  (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
         LINK_FLAGS      "-nodefaultlibs"
         LINKER_LANGUAGE C)
     if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86|i.86")
-        add_definitions(-mfpmath=sse -msse2)
+        add_definitions(-mfpmath=sse)
     endif()
     target_link_libraries(graphite2 c)
     include(Graphite)
