diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index e91d9e1..cc0a48f 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -12,11 +12,11 @@ if(A2X)
     if(DBLATEX)
         add_custom_command(OUTPUT ${PROJECT_BINARY_DIR}/manual.pdf
                            DEPENDS ${PROJECT_SOURCE_DIR}/[a-z]*.asc
-                           COMMAND ${A2X} --icons -D ${PROJECT_BINARY_DIR} ${PROJECT_SOURCE_DIR}/manual.asc)
+                           COMMAND ${A2X} -D ${PROJECT_BINARY_DIR} ${PROJECT_SOURCE_DIR}/manual.asc)
         add_custom_command(OUTPUT ${PROJECT_BINARY_DIR}/GTF.pdf
                            DEPENDS ${PROJECT_SOURCE_DIR}/GTF.asc
                                    ${PROJECT_SOURCE_DIR}/OpCodes.asc
-                           COMMAND ${A2X} --icons -D ${PROJECT_BINARY_DIR} ${PROJECT_SOURCE_DIR}/GTF.asc)
+                           COMMAND ${A2X} -D ${PROJECT_BINARY_DIR} ${PROJECT_SOURCE_DIR}/GTF.asc)
         set(DOC_DEPENDS ${PROJECT_BINARY_DIR}/manual.pdf ${PROJECT_BINARY_DIR}/GTF.pdf)
     endif()
     add_custom_command(OUTPUT ${PROJECT_BINARY_DIR}/manual.html
