--- a/python/graphite2/__init__.py
+++ b/python/graphite2/__init__.py
@@ -32,15 +32,8 @@
                     CFUNCTYPE, POINTER, Structure)
 
 
-libpath = os.environ.get('PYGRAPHITE2_LIBRARY_PATH',
-                         ctypes.util.find_library("graphite2"))
-if libpath is None:
-    # find wheel's library
-    wheel = os.path.dirname(__file__)
-    if os.name == 'nt':
-        libpath = os.path.join(wheel, 'bin', 'graphite2.dll')
-    else:
-        libpath = os.path.join(wheel, 'lib', 'libgraphite2.so')
+libpath = "libgraphite2.so.3"
+
 gr2 = ctypes.cdll.LoadLibrary(libpath)
 
 
