--- a/Graphite.cmake
+++ b/Graphite.cmake
@@ -79,7 +79,7 @@
     endif (EXISTS ${PROJECT_SOURCE_DIR}/standards/${TESTNAME}${CMAKE_SYSTEM_NAME}.log)
     if (NOT GRAPHITE2_NFILEFACE)
         add_test(NAME ${TESTNAME} COMMAND $<TARGET_FILE:gr2fonttest> -trace ${PROJECT_BINARY_DIR}/${TESTNAME}.json -log ${PROJECT_BINARY_DIR}/${TESTNAME}.log ${PROJECT_SOURCE_DIR}/fonts/${FONTFILE} -codes ${ARGN})
-        set_tests_properties(${TESTNAME} PROPERTIES TIMEOUT 3)
+        set_tests_properties(${TESTNAME} PROPERTIES TIMEOUT 120)
         add_test(NAME ${TESTNAME}Output COMMAND ${CMAKE_COMMAND} -E compare_files ${PROJECT_BINARY_DIR}/${TESTNAME}.log ${PROJECT_SOURCE_DIR}/standards/${TESTNAME}${PLATFORM_TEST_SUFFIX}.log)
         if ((NOT GRAPHITE2_NTRACING) AND PYTHONINTERP_FOUND)
             add_test(NAME ${TESTNAME}Debug COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/jsoncmp ${PROJECT_BINARY_DIR}/${TESTNAME}.json ${PROJECT_SOURCE_DIR}/standards/${TESTNAME}.json)
@@ -96,7 +96,7 @@
     endif (EXISTS ${PROJECT_SOURCE_DIR}/standards/${TESTNAME}${CMAKE_SYSTEM_NAME}.log)
     if (NOT GRAPHITE2_NFILEFACE)
         add_test(NAME ${TESTNAME} COMMAND $<TARGET_FILE:gr2fonttest> -log ${PROJECT_BINARY_DIR}/${TESTNAME}.log ${PROJECT_SOURCE_DIR}/fonts/${FONTFILE})
-        set_tests_properties(${TESTNAME} PROPERTIES TIMEOUT 3)
+        set_tests_properties(${TESTNAME} PROPERTIES TIMEOUT 120)
         add_test(NAME ${TESTNAME}Output COMMAND ${CMAKE_COMMAND} -E compare_files ${PROJECT_BINARY_DIR}/${TESTNAME}.log ${PROJECT_SOURCE_DIR}/standards/${TESTNAME}${PLATFORM_TEST_SUFFIX}.log)
         set_tests_properties(${TESTNAME}Output PROPERTIES DEPENDS ${TESTNAME})
     endif (NOT GRAPHITE2_NFILEFACE)
